README.md

```diff
@@ -1,6 +1,197 @@
+
+At this point, the user will observe that their browser has navigated to `https://www.example.com/` (e.g., via changes to the URL bar contents and back/forward UI). Since `https://example.com/` was already loaded in the portal context, this navigation will occur seamlessly and instantly, without a network round-trip or document re-initialization.
+
+For more advance use cases, the `https://www.example.com/` document can react to activation, using the `portalactivate` event. It can use this event to adapt itself to its new context, and even to adopt its *predecessor* (the document which previously occupied the tab) into a new portal context.
+
+```js
+window.addEventListener('portalactivate', e => {
+  document.body.classList.add('displayed-fully');
+  document.requestStorageAccess();
+
+  let predecessor = e.adoptPredecessor(document);
+  console.assert(predecessor instanceof HTMLPortalElement);
+  document.body.appendChild(predecessor);
+});
+```
+
+## Use cases
+
+_See the ["Key Scenarios"](./key-scenarios.md) document for more detail on each of these, including visualizations._
+
+- **Pre-rendering**: by loading a page in a hidden portal, it is possible to pre-render the destination page. Then, activating the portal will instantly display the pre-rendered document in the same browsing context.
+
+  This requires some care and cooperation from both sides to deal with the restrictions on portaled content, and the transition between portaled and activated states.
+
+- **Navigation transitions**: pre-rendering opens the door for more elaborate transitions, by displaying the portal in some form, animating it (using resizing, translations, etc.) until it occupies the full viewport, then finally activating the portal to perform the instant navigation.
+
+- **Aggregation**: multiple portals on the same page can be used to create more elaborate experiences, where the user chooses which portal to activate. This category of use cases includes cases like a news reader, a shopping site, an infinite scrolling list of articles, etc. By using a portal instead of (or in addition to) a link, the aggregated content has the opportunity to display a preview, and to benefit from pre-rendering and navigation transitions.
+
+  Additionally, by using the ability to adopt the predecessor during the `portalactivate` event, more complicated interactions between the aggregator and the aggregated content can be built, such as retaining a portion of the shopping site or article-list UI in a portal even after navigating to an individual page.
+
+- **"Better iframe"**: portals encompass some, but not all, of the use cases for iframes. And they do so in a way that is better for users, in terms of security and privacy. They also remove a lot of the legacy baggage and sharp edges that come with iframes, making them easier to use for web developers. So we anticipate many of the current cases for iframes, such as ads, being able to be replaced with portals.
```

Comment:
"such as ads" might not be super-accurate here. In general it'd be good to get a better sense of which iframe use cases can be done via portals. (That's one of the TODOs below.)

Reply:
I'm not sure if you'd use a portal for things that don't need to activate.

Reply:
I mean, ads need to be navigated to, and activating them seems like a nicer way of navigating? I'm not sure though.

Reply:
Are there other cases to mention where people might want to use portals instead of iframes? Maybe same-origin ones?